core/src/main/scala/org/apache/spark/scheduler/local/LocalSchedulerBackend.scala

```diff
@@ -162,7 +162,13 @@ private[spark] class LocalSchedulerBackend(
 
   override def applicationId(): String = appId
 
-  override def maxNumConcurrentTasks(): Int = totalCores / scheduler.CPUS_PER_TASK
+  // Doesn't support different ResourceProfiles yet
+  // so we expect all executors to be of same ResourceProfile
+  override def maxNumConcurrentTasks(rp: ResourceProfile): Int = {
+    val cpusPerTask = rp.taskResources.get(ResourceProfile.CPUS)
+      .map(_.amount.toInt).getOrElse(scheduler.CPUS_PER_TASK)
+    totalCores / cpusPerTask
```

Comment:
Curious, in future are we planning to support resource profiles for local mode ? With local gpu for example ?

Reply:
not currently planned but someone did just say they were going to add the generic resource scheduling work in local mode.  I think for testing purposes so they can keep production code same. did you have specific use case or concern?

Reply:
If possible, feature parity in whatever practical/constrained way of local mode with cluster mode would be nice - it allows for an quick prototyping/testing before launching on a cluster.
Ofcourse, not coupling with this work - but particularly for rapid prototyping, local mode is invaluable.

Reply:
yeah the main issue right now is this is tied to dynamic allocation, so we would just need to figure out we would want that to behave.